aws/resource_aws_xray_sampling_rule.go

```diff
@@ -154,13 +159,30 @@ func resourceAwsXraySamplingRuleRead(d *schema.ResourceData, meta interface{}) e
 	d.Set("url_path", samplingRule.URLPath)
 	d.Set("version", samplingRule.Version)
 	d.Set("attributes", aws.StringValueMap(samplingRule.Attributes))
-	d.Set("arn", samplingRule.RuleARN)
+	d.Set("arn", arn)
+
+	tags, err := keyvaluetags.XrayListTags(conn, arn)
+	if err != nil {
+		return fmt.Errorf("error listing tags for Xray Sampling group (%q): %s", d.Id(), err)
+	}
+
+	if err := d.Set("tags", tags.IgnoreAws().IgnoreConfig(ignoreTagsConfig).Map()); err != nil {
+		return fmt.Errorf("error setting tags: %w", err)
+	}
 
 	return nil
 }
 
 func resourceAwsXraySamplingRuleUpdate(d *schema.ResourceData, meta interface{}) error {
 	conn := meta.(*AWSClient).xrayconn
+
+	if d.HasChange("tags") {
+		o, n := d.GetChange("tags")
+		if err := keyvaluetags.XrayUpdateTags(conn, d.Get("arn").(string), o, n); err != nil {
+			return fmt.Errorf("error updating tags: %w", err)
+		}
+	}
+
 	samplingRuleUpdate := &xray.SamplingRuleUpdate{
```

Comment:
Nit: For `tags` only updates -- we should skip the `UpdateSamplingRule` API call below. I really need to get this "HasChangesOtherThan" thing merged upstream 😅

Reply:
I was wondering if a should handle this, NP. coming right up.

Reply:
Done, funny enough it takes longer to update tags then other arguments

Reply:
This may need to be split more neatly to see if things can partially updated but left as is